maplibre/src/headless/system.rs

```diff
@@ -55,10 +55,12 @@ impl System for WriteSurfaceBufferSystem {
                 let padded_buffer = buffer_slice.get_mapped_range();
 
                 if self.write_to_disk {
-                    buffered_texture.write_png(
-                        &padded_buffer,
-                        format!("frame_{}.png", current_frame).as_str(),
-                    );
+                    buffered_texture
+                        .write_png(
+                            &padded_buffer,
+                            format!("frame_{}.png", current_frame).as_str(),
+                        )
+                        .expect("Could save frame to disk");
```

Comment:
I see two possible solutions here:

- Accept that some excepts are fine. For instance wgpu is also panicing for some inputs.
- Systems should be able to return errors, alternatively, the errors need to be sent somewhere.

Reply:
I would say that on a longer run, it will be nicer to move the errors up and that system return errors.

But the current state seems acceptable to me: if a headless render can’t save to disk, it’s ok to fail. (but maybe I’m too much baby steps oriented?)